### PR TITLE
Example of Ensemble Kalman filter with Out-Of-Sequence measurements

### PR DESCRIPTION
**Pull request by A-acuto**
This PR adds a simple example of using a Kalman Filter with measurements with a fixed delay from a sensor.

As referenced by #940 and #939 , this is a series of examples that shows different methods to deal with OOSM. 
The example has been modified by adding a EK filter for comparison.

### CODE REVIEW

**Review by spike-dstl — CHANGES_REQUESTED** [review]
Grammatical suggestions to support email sent 19/02/2024.

File: docs/examples/KalmanFilterOOSMExample.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+Kalman filter with Out-of-Sequence measurements
+===============================================
+"""
+
+# %%
+# In other examples we have shown how to deal with out-of-sequence measurements (OOSM)
+# with methods like using inverse-time dynamics or creating a buffer where store and re-order
+# the measurements according to their delay.
+# In this example we present a method on how to deal with OOS measurements using Kalman
+# filters, my making the assumption that the delay of the delayed measurements is known, and constant.
```
Comment (spike-dstl) [inline docs/examples/KalmanFilterOOSMExample.py:15]:
```suggestion
# filters, by making the assumption that the delay of the delayed measurements is known, and constant.
```

**Review by spike-dstl — APPROVED** [review]
One final spelling mistake correction. Approving with assumption that this will be fixed.

File: docs/examples/KalmanFilterOOSMExample.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+# The problem of OOS measurements is significant in real-world applications
+# where data from different sources can have some delays and different timesteps
+# (e.g., two sensors observing a target) due to systems configuration and different processing
+# chain length.
+#
+# In the literature, there are examples (e.g. [#]_) on how to deal with such time-delays and
+# uncertain timesteps.
```
Comment (nperree-dstl) [inline docs/examples/KalmanFilterOOSMExample.py:22]:
```suggestion
# In the literature, there are examples on how to deal with such time-delays and
# uncertain timesteps [#]_.
```

**Review by nperree-dstl — APPROVED** [review]
Only a minor suggestion, otherwise looks good!

**Pull request merged**